### DISCUSSION TITLE
Design choices related to jvp/linearize/...

### DISCUSSION THREAD

**Discussion by pharringtonp19** [jax-ml/jax#8050] [Q&A] [answered]
I should start off by saying that I enjoy working with `jax.jvp` and `jax.linearize`, so the following is not a complaint but rather a question about the design choice of these transformations.

After reading this [paper](https://arxiv.org/pdf/2105.09469.pdf), I am a bit surprised about the following example which I took from [the docs](https://jax.readthedocs.io/en/latest/jax.html#jax.linearize)

```
y, f_jvp = jax.linearize(f, x)
```

I would have thought that the following would have been the design choice. Am I missing something obvious? Thanks
```
y, f_jvp = jax.linearize(f)(x)
```

**Comment by AntixK** [comment, marked as answer]
Actually, the current design choice makes more sense because the linearization happens at the primal point `x`. Since `f` can be any function, it is crucial to linearize only at the points of concern i.e. `x`, 

In your case, it would seem that the function is first linearized and then evaluated at `x`, which is certainly not the case.
